### PR TITLE
feat(cb2-6869): add a flag to track record visibility in vta

### PR DESCRIPTION
**Pull request by tahayazici**
## Prevent a Tech Record from being testable

- Add a new flag to track record visibility on VTA

[CB2-6869](https://dvsa.atlassian.net/browse/CB2-6869)

### CODE REVIEW

**Review by aaron-downing — CHANGES_REQUESTED** [review]
Would be good if you could fix this integration test with this PR, I realise the issue isn't to do with the functionality you've implemented
image

**Review by andrewf-bjss — COMMENTED** [review]
Do we need to do anything with this new flag?

**Comment by andrewf-bjss** [comment]
I think we might have to check this new flag in the `recordCompleteness` function to hide the vehicle from VTA

**Comment by tahayazici** [comment]
@aaron-downing I would appreciate it if we create a new PR to fix that integration test as it is unrelated and I do not know what it is. Plus, as you saw yesterday, we had difficulties running the integration tests on my machine.

The UI work relies on this, so it would be good to get this in.

**Comment by tahayazici** [comment]
@aaron-downing The integration has been fixed by @gjulien-bjss.

**Review by andrewf-bjss — APPROVED** [review]

File: src/utils/record-completeness/ComputeRecordCompleteness.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ export function computeRecordCompleteness(vehicle: Vehicle): string {
     throw new HTTPError(400, ERRORS.SYSTEM_NUMBER_GENERATION_FAILED);
   }
 
+  if (vehicle.techRecord[0].hiddenInVta) {
```
Comment (aaron-downing) [inline src/utils/record-completeness/ComputeRecordCompleteness.ts:22]: Is this method only called on the PUT updateTechRecords endpoint? Just wondering if there's ever an instance where techRecord has a length greater than 1 and this hard coded indexing might cause issues?

Comment (tahayazici) [reply src/utils/record-completeness/ComputeRecordCompleteness.ts:22]: All utility methods use the same approach to get the `techRecord` so I followed suit.

This is called on POST and PUT.

Comment (tahayazici) [reply src/utils/record-completeness/ComputeRecordCompleteness.ts:22]: Sure, I posted this PR on the `Code Reviews` channel.

**Review by Ian-Hodges — APPROVED** [review]

File: src/utils/record-completeness/ComputeRecordCompleteness.ts (hunk 1 above)
Comment (Ian-Hodges) [reply src/utils/record-completeness/ComputeRecordCompleteness.ts:22]: Ignore my comment above. I was thinking from a GET point of view, which this is not the case.

**Review by gjulien-bjss — APPROVED** [review]

**Review by aaron-downing — APPROVED** [review]

**Pull request merged**